modules/misc/gtk.nix

```diff
@@ -166,14 +166,36 @@ in {
     home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
       ++ optionalPackage cfg.iconTheme;
 
-    home.file.${cfg2.configLocation}.text =
-      concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
-      + cfg2.extraConfig;
+    home.file.${cfg2.configLocation} = {
+      text = concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
+        + cfg2.extraConfig;
+      onChange = mkIf (cfg.theme != null)
+        (let xsettingsd = "${pkgs.xsettingsd}/bin/xsettingsd";
+        in ''
+          DISPLAY=$(grep "declare -x DISPLAY=" \
+          "''${XDG_RUNTIME_DIR:-/run/user/$UID}/env-vars" \
```

Comment:
What is this `env-vars` file?

Reply:
I was trying to get DISPLAY from somewhere without pinning it, and this seemed to work.
I've since migrated to flakes and it seems it doesn't anymore. So not very reliable (not sure what used to write it in my system).

When #2147 is merged this won't be needed anymore, as DISPLAY will be available.

I have a couple of changes i didn't push to this branch, including removing that. Gonna merge them in now